dtfs-central-api/src/repositories/utilisation-reports-repo/utilisation-report.repo.ts

```diff
@@ -236,4 +236,24 @@ export const UtilisationReportRepo = SqlDbDataSource.getRepository(UtilisationRe
       },
     });
   },
+
+  /**
+   * Finds the years where the bank with the supplied id submitted
+   * a utilisation report
+   * @param bankId - The bank id
+   * @returns A set of years where the bank submitted a report
+   */
+  async findReportingYearsByBankId(bankId: string): Promise<Set<number>> {
+    const utilisationReports = await this.find({
+      where: {
+        bankId,
+        status: Not('REPORT_NOT_RECEIVED'),
+      },
+      select: ['reportPeriod'],
+    });
+    return utilisationReports.reduce(
+      (uniqueReportPeriodYears, { reportPeriod }) => uniqueReportPeriodYears.add(reportPeriod.start.year).add(reportPeriod.end.year),
+      new Set<number>(),
+    );
+  },
```

Comment:
This is becoming large, nearly 300 lines, anyway we can split up / extract and import? Probably not for this PR.

Reply:
I don't think there is anything wrong with the size of this file (see the various `api.js` files for instance). The reason we should keep these methods in this file aside from anything else is that they are appending to the base repository api through `.extend`, which keeps our binding of `this` consistent. If we try to move the functions outside of this file and them import them, we start having to worry about binding `this` to the correct value, which we don't want to do